examples/tutorials/advanced/dataset_on_surface.py

```diff
@@ -0,0 +1,164 @@
+
+It can be visually appealing to "drape" a data set over a topographic surface.
+This can be accomplished using the ``drapegrid`` parameter of
+:meth:`pygmt.Figure.grdview`.
+
+This tutorial consists of two examples:
+
+1. Draping a grid
+
+2. Draping an image
+"""
+
+# %%
+
+# Load the required packages
+import pygmt
+import rasterio
+import xarray as xr
+
+# %%
+# 1. Drapping a grid
+# ------------------
+#
+# In the first example, the seafloor crustal age is plotted with color-coding
+# on top of the topographic map of an area of the Mid-Atlantic Ridge.
+
+# Define study area
+region_2d = [-50, 0, 36, 70]  # [lon_min, lon_max, lat_min, lat_max]
+region_3d = [*region_2d, -5961, 3366.5]  # Append [z_min, z_max] to this list
+perspective = [157.5, 30]  # Define azimuth, elevation for the 3-D plot
+
+# -----------------------------------------------------------------------------
+# Create new Figure instance
+fig = pygmt.Figure()
+
+# Download elevation and crustal age grids for the study region with a
+# resolution of 3 arc-minutes and load them into xarray.DataArrays
+grd_relief = pygmt.datasets.load_earth_relief(resolution="03m", region=region_2d)
+grd_age = pygmt.datasets.load_earth_age(resolution="03m", region=region_2d)
+
+# Set up colormap for curstal age
+pygmt.config(COLOR_NAN="lightgray")
+pygmt.makecpt(cmap="batlow", series=[0, 200, 1], reverse=True, overrule_bg=True)
+
+fig.grdview(
+    projection="M12c",  # Mercator projection with a width of 12 centimeters
+    region=region_3d,
+    grid=grd_relief,  # Use elevation grid for z values
+    drapegrid=grd_age,  # Use crustal age grid for color-coding
+    cmap=True,
+    surftype="i",  # Create an image plot
+    # Use an illumination from the azimuthal directions 0° (north) and 270°
+    # (west) with a normalization via a cumulative Laplace distribution for
+    # the shading
+    shading="+a0/270+ne0.6",
+    perspective=perspective,
+    zsize="1.5c",
+    plane="+gdarkgray",
+    frame=True,
+)
+
+# -----------------------------------------------------------------------------
+# Add colorbar for curstal age
+fig.colorbar(frame=["x+lseafloor crustal age", "y+lMyr"], position="+n")
+
+# Show figure
+fig.show()
+
+
+# %%
+# 2. Draping an image
+# -------------------
+#
+# In the second example, the flag of Europe is plotted on top of a topographic
+# map of northwest Europe. This example is modified from GMT example 32 available
+# at https://docs.generic-mapping-tools.org/latest/gallery/ex32.html#example-32.
+
+# Define study area
+region_2d = [3, 9, 50, 54]  # [lon_min, lon_max, lat_min, lat_max]
+region_3d = [*region_2d, -10, 900]  # Append [z_min, z_max] to this list
+perspective = [157.5, 30]  # Define azimuth, elevation for the 3-D plot
+
+# Coordinates and names of cities
+lons_cities = [7.10, 4.35, 5.69]  # degrees East
+lats_cities = [50.73, 50.85, 50.85]  # degrees North
+text_cities = ["Bonn", "Bruxelles", "Maastricht"]
```

Comment:
Instead of creating three lists, a pandas.DataFrame object is more Pythonic and easy to read:
```
cities = pd.DataFrame({
    "longitude": [7.10, 4.35, 5.69], 
    "latitude": [50.73, 50.85, 50.85], 
    "name": ["Bonn", "Bruxelles", "Maastricht"]
})
```

Reply:
Yep, also thought about this, to avoid mixing up the values. Updated in commit https://github.com/GenericMappingTools/pygmt/pull/3316/commits/9976c5b3297293420b4cc26cf63ccd80c856bab2.